app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/overview/TravelCertificateOverview.kt

```diff
@@ -0,0 +1,144 @@
+package com.hedvig.android.feature.travelcertificate.ui.overview
+
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.hedvig.android.core.designsystem.component.button.HedvigContainedButton
+import com.hedvig.android.core.designsystem.preview.HedvigPreview
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.core.ui.appbar.m3.TopAppBarActionType
+import com.hedvig.android.core.ui.clearFocusOnTap
+import com.hedvig.android.core.ui.infocard.DrawableInfoCard
+import com.hedvig.android.core.ui.infocard.VectorInfoCard
+import com.hedvig.android.core.ui.scaffold.HedvigScaffold
+import com.hedvig.android.feature.travelcertificate.data.TravelCertificateUri
+import com.hedvig.android.feature.travelcertificate.data.TravelCertificateUrl
+import com.hedvig.android.feature.travelcertificate.ui.FullScreenLoading
+import com.hedvig.android.feature.travelcertificate.ui.SomethingWrongInfo
+import hedvig.resources.R
+
+@Composable
+internal fun TravelCertificateOverviewDestination(
+  travelCertificateUrl: TravelCertificateUrl,
+  viewModel: TravelCertificateOverviewViewModel,
+  navigateUp: () -> Unit,
+  onShareTravelCertificate: (TravelCertificateUri) -> Unit,
+) {
+  val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+  TravelCertificateOverview(
+    travelCertificateUrl = travelCertificateUrl,
+    onDownloadCertificate = { viewModel.emit(TravelCertificateOverviewEvent.OnDownloadCertificate(it)) },
+    navigateBack = navigateUp,
+    onShareTravelCertificate = onShareTravelCertificate,
+    uiState = uiState,
+  )
+}
+
+@Composable
+internal fun TravelCertificateOverview(
+  travelCertificateUrl: TravelCertificateUrl,
+  onDownloadCertificate: (TravelCertificateUrl) -> Unit,
+  navigateBack: () -> Unit,
+  onShareTravelCertificate: (TravelCertificateUri) -> Unit,
+  uiState: TravelCertificateOverviewUiState,
+) {
+  when (uiState) {
+    TravelCertificateOverviewUiState.Failure -> {
+      HedvigScaffold(
+        navigateUp = { navigateBack() },
+        topAppBarActionType = TopAppBarActionType.CLOSE,
+        modifier = Modifier.clearFocusOnTap(),
```

Comment:
```suggestion
        topAppBarActionType = TopAppBarActionType.CLOSE,
```
I think in this scenario there is nothing that would show a text field input, therefore there would be nothing to clear the focus from.
With that said, I don't think just having the modifier there breaks anything, or is bad in any way, but it won't really do something in this scenario.
I would be okay either way, but I know if I was around here and was moving any code around I would probably instinctively just remove this.